bundles/core/src/main/java/com/adobe/cq/commerce/core/components/internal/models/v1/productlist/ProductListImpl.java

```diff
@@ -97,19 +96,15 @@ private void initModel() {
 
         // get GraphQL client and query data
         if (categoryId != null) {
-            GraphqlClient client = resource.adaptTo(GraphqlClient.class);
-            this.category = getCategory(client, categoryId);
+            client = new MagentoGraphqlClient(resource);
```

Comment:
Older bug: missing error handling when client is null. 
Now client is never null but it can be invalid.